### PR TITLE
Mh/travis ci/new

### PR DESCRIPTION
**Pull request by bleehu**
Rewrote most of CXDocs to have Travis Continuous Integration. This will make sure that each future change doesn't break the build. Which should in turn make it easier for more people to work on it.

Also rewrote start.py to be the app.py that flask expects. This will let us do things like launch using the flask command, not bare python. 

This change currently isn't documented, and currently breaks the old arguments workflow that allows for signing in; those changes are still pending.

### CODE REVIEW

File: server/app.py [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+from mission import Mission #Mission is a custom data typ that we made to organize mission info on the backend.
+import pdb  #Python Debuger is what I use to fix borked code. It should not be called in production EVER!
+#but it's very helpful when being run locally.
+
+import psycopg2 #psycopg2 lets us make posgres SQL calls from python. That lets us store things in databases
+import os   #we need os to read and write files as well as to make our filepaths relative.
+import logging #When we aren't running locally, we need the server to log what's happening so we can see any
+#intrusions or help debug why it's breaking if it does so. This module handles that beautifully.
+from security import security #our custom code that handles common security tasks like SQL sanitization
+import xml.etree.ElementTree #Sometimes we write or read things in XML. This does that well.
```
Comment (Turtlelord26) [inline server/app.py]: Appreciating the import comments a lot.

File: server/app.py [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+        if config.has_option('Parser', 'carbines_filepath'):
+            itemsDocs.append(('Carbines and Assault Rifles','/docs/carbines'))
+
+        if config.has_option('Parser', 'long_rifles_filepath'):
+            itemsDocs.append(('Long Rifles and DMRs','/docs/longRifles'))
+
+        if config.has_option('Parser', 'machineguns_filepath'):
+            itemsDocs.append(('Machine Guns and Rocket Launchers','/docs/machineguns'))
+
+        if config.has_option('Parser', 'explosives_filepath'):
```
Comment (Turtlelord26) [inline server/app.py]: Missing some weapon types?

File: server/app.py [hunk 3]
```diff
@@ -0,0 +1,551 @@
...
+    initialize_enemies(config, log)
+    initialize_characters(config, log)
+    (username, password, host) = get_env_vars()
+    app.config['username'] = username
+    app.config['password'] = password
+    app.config['ip_address'] = host
+
+    security.initialize(username, password, log)
+
+    app.secret_key = '$En3K9lEj8GK!*v9VtqJ' #todo: generate this dynamically
```
Comment (Turtlelord26) [inline server/app.py]: Would check, but screen too tiny. Make sure is an Issue pls.

File: README.md [hunk 4]
```diff
@@ -24,62 +25,158 @@ Compound X rules displayed as web pages. Written in Python 2.7, served with Flas
...
+
+A [lock file](https://askubuntu.com/questions/530598/what-is-the-purpose-of-lock-file)
+for your pipfile. It also holds security hashes to make sure that someone isn't
+poisoning your python libraries.
+
+## Readme.md
+
+A markdown file that shows easy-to-read instructions for using and contributing
+to CXDocs. Readme files are displayed on github direcory pages automatically,
+and markedown readme files are prettier than text files. What you're reading
```
[the remark this answers is not in the document]
Comment (bleehu) [reply README.md]: Just to make it confusing, as specific markup language called Markedown https://guides.github.com/features/mastering-markdown/

File: server/app.py (hunk 2 above)
Comment (bleehu) [reply server/app.py]: Several of the documents have two types of weapons in them each; such as SMGs and Shotguns.

**Pull request merged**